From akpm@osdl.org Thu Jan  5 22:54:11 2006
Message-Id: <200601060650.k066omZZ008821@shell0.pdx.osdl.net>
From: Andrew Morton <akpm@osdl.org>
Subject: USB: usbip build fix
To: greg@kroah.com
Cc: akpm@osdl.org, taka-hir@is.naist.jp
Date: Thu, 05 Jan 2006 22:50:36 -0800


From: Andrew Morton <akpm@osdl.org>

Fix this:

drivers/usb/ip/usbip_common.c: In function `socket_to_addrstr':
drivers/usb/ip/usbip_common.c:582: dereferencing pointer to incomplete type
drivers/usb/ip/usbip_common.c:582: dereferencing pointer to incomplete type
drivers/usb/ip/usbip_common.c:582: dereferencing pointer to incomplete type
drivers/usb/ip/usbip_common.c:582: dereferencing pointer to incomplete type

And this:

drivers/usb/ip/usbip_common.h: At top level:
drivers/usb/ip/usbip_common.c:441: warning: `usbip_dump_urb' defined but not used

Cc: Takahiro Hirofuchi <taka-hir@is.naist.jp>
Signed-off-by: Andrew Morton <akpm@osdl.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/usb/ip/usbip_common.c |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

--- gregkh-2.6.orig/drivers/usb/ip/usbip_common.c
+++ gregkh-2.6/drivers/usb/ip/usbip_common.c
@@ -23,6 +23,7 @@
 #include <linux/kernel.h>
 #include <linux/file.h>
 #include <linux/tcp.h>
+#include <linux/in.h>
 #include "usbip_common.h"
 
 static void usbip_dump_buffer(char *, int size);
@@ -254,7 +255,7 @@ static void usbip_dump_pipe(unsigned int
 
 }
 
-
+#if 0
 static void usbip_dump_usb_device(struct usb_device *dev)
 {
 	if (dev == NULL) {
@@ -466,7 +467,7 @@ static void usbip_dump_urb (struct urb *
 	printk("   context               :%p\n", purb->context);
 	printk("   complete              :%p\n", purb->complete);
 }
-
+#endif
 
 void setquickack(struct socket *socket)
 {
